Reapply changes by @onei from VOLDEV-145

mixth-sense opened the pull request:
# VOLDEV-145 & SUS-28 & XW-396
- https://wikia-inc.atlassian.net/browse/VOLDEV-145
- https://wikia-inc.atlassian.net/browse/XW-396
- https://wikia-inc.atlassian.net/browse/SUS-28
## Description

It was unable to simply merge @onei's PR https://github.com/Wikia/app/pull/8755 because of the changes we have made in the meantime.

So I examined file by file all the changes made by @onei while implementing VOLDEV-145 and re-applied it manually onto our fresh dev. There were 27 files in the original PR and as you can see, I named my commits accordingly.
### Removed files

FIles listed in this section were simply removed by VOLDEV-145
- `PageHeader_EditBox.php`
### Clean files (only @onei's changes):

Files listed in this section remained unchanged within the timeframe between #8755 and now and can be safely replaced:
- `ForumHooksHelper.class.php`
- ~~`LicensedVideoSwapHooksHelper.class.php`~~ (removed in the meantime along with the entire extension)
- `Oasis_setup.php`
- ~~`PageShare.js`~~ — incorrectly changed, we keep the original version
- `PageShare.scss`
- `PlacesHooks.class.php`
- `SpecialVideos.setup.php`
- `WikiaHubsV3.scss`
- `PageHeader.scss`
- `BodyController.class.php`
- `PageHeader_Corporate.php`
- `PageHeader_EditPage.php`
- `PageHeader_Hubs.php`
- `WikiHeader_Index.php`
### Changed files (by @onei and Wikia in the meantime)

Files listed in this section have been changed by VOLDEV-145 and also changed by Wikia within the timeframe between #8755 and now. I have examined them and applied bot @onei's and Wikia's changes manually.
- `WallHooksHelper.class.php`
- `WikiaNewFilesHooks.php`
- `Article.php`
- `DefaultSettings.php`
- `SpecialVideosHooks.php`
- `PageHeaderController.class.php`
- `Body_Index.php`
- `PageHeader_Index.php`
### New files (added by @onei)

Files listed in this section have been added by @onei while implementing VOLDEV-145 and can be safely added to our fresh dev.
- `WikiaNewFilesHooks.php` — then renamed to `WikiaNewFilesHooks.class.php`
- `ContributeController.class.php` — originally created by @onei
- `ContributeModule.scss` — originally created by @onei
- `Contribute_index.php` — originally created by @onei
## Code Reviewers
- @onei — please, review the code as you are the author of the changes.
- @Warkot — in case you still remember the context, your pair of eyes would be helpful
- @Wikia/sustaining-team

Commit:
01 of 27 - clean - ForumHooksHelper.class.php

Commit:
02 of 27 - clean - LicensedVideoSwapHooksHelper.class.php

Commit:
03 of 27 - clean - Oasis_setup.php

Commit:
05 of 27 - clean - PageShare.scss

Commit:
06 of 27 - clean - PlacesHooks.class.php

Commit:
07 of 27 - clean - SpecialVideos.setup.php

Commit:
08 of 27 - clean - WikiaHubsV3.scss

Commit:
09 of 27 - clean - PageHeader.scss

Commit:
10 of 27 - clean - BodyController.class.php

Commit:
11 of 27 - clean - PageHeader_Corporate.php

Commit:
12 of 27 - removed - PageHeader_EditBox.php

Commit:
13 of 27 - clean - PageHeader_EditPage.php

Commit:
14 of 27 - clean - PageHeader_Hubs.php

Commit:
15 of 27 - clean - WikiHeader_Index.php

Commit:
-- of 27 - removed - LicensedVideoSwapHooksHelper.class.php

Commit:
16 of 27 - changed - WallHooksHelper.class.php

Commit:
17 of 27 - added - WikiaNewFilesHooks.php

Commit:
18 of 27 - changed - WikiaNewFiles_setup.php

Commit:
19 of 27 - changed - Article.php

Commit:
20 of 27 - changed - DefaultSettings.php

Commit:
21 of 27 - added - ContributeController.class.php

Commit:
22 of 27 - added - ContributeModule.scss

Commit:
23 of 27 - added - Contribute_index.php

Commit:
24 of 27 - added - SpecialVideosHooks.php

Commit:
-- of 27 - fix naming conventions

Commit:
25 of 27 - changed - PageHeaderController.class.php

Commit:
Merge branch 'dev' into SUS-28-dev-VOLDEV-145-reapply-changes-by-onei

Commit:
-- of 27 - changed - PageHeaderController.class.php

Commit:
26 of 27 - changed - Body_Index.php

Commit:
-- of 27 - changed - Body_Index.php

Commit:
27 of 27 - changed - PageHeader_Index.php

Commit:
Merge branch 'dev' into SUS-28-dev-VOLDEV-145-reapply-changes-by-onei

Commit:
Merge branch 'dev' into SUS-28-dev-VOLDEV-145-reapply-changes-by-onei

Commit:
Merge branch 'dev' into SUS-28-dev-VOLDEV-145-reapply-changes-by-onei

Commit:
Merge branch 'dev' into SUS-28-dev-VOLDEV-145-reapply-changes-by-onei

kvas-damian commented on extensions/wikia/Forum/ForumHooksHelper.class.php:
```diff
@@ -237,20 +237,31 @@ static function onGetUserPermissionsErrors( Title &$title, User &$user, $action,
+
+		// check namespace(s)
+
 		if ( $ns == NS_FORUM || $ns == NS_FORUM_TALK ) {
```
While touching this method maybe we could use strict comparison here?

michalroszka replied:
:white_check_mark: addressed in 1cb5b960e7626840d729b87dc9e6005494890f12

kvas-damian commented on extensions/wikia/Places/PlacesHooks.class.php:
```diff
@@ -7,41 +7,50 @@ static public function setModelToSave( PlaceModel $model ){
+						'href' => '#',
+						'text' => wfMessage( 'places-category-switch' )->escaped()
+					],
+					'class' =>  !$isGeotaggingEnabled ? $commonClasses . $disabled : $commonClasses,
```
Maybe little bit simpler version: `$commonClasses . (!$isGeotaggingEnabled ? $disabled : '')`?

michalroszka replied:
:warning: addressed in 1cb5b96, needs whitespace fixes to meet MW coding conventions

onei replied:
Whitespace fixed in [de16f7f](https://github.com/Wikia/app/commit/de16f7f610f2d01ba4f8ace5263b7e4160cbf995)

kvas-damian commented on extensions/wikia/Places/PlacesHooks.class.php:
```diff
@@ -7,41 +7,50 @@ static public function setModelToSave( PlaceModel $model ){
+						'href' => '#',
+						'text' => wfMessage( 'places-category-switch-off' )->escaped()
+					],
+					'class' => $isGeotaggingEnabled ? $commonClasses . $disabled : $commonClasses,
```
This class name is computed 2 times. Maybe we can move it up and do it only once?

kvas-damian commented on extensions/wikia/SpecialVideos/SpecialVideosHooks.class.php:
```diff
@@ -0,0 +1,49 @@
+			$app->wg->Title->isSpecial( 'Videos' ) &&
+			$app->wg->User->isAllowed( 'videoupload' )
+		) {
+			$text = Html::element(
```
Variable name is little confusing. It's not `text` when we put `img` there.

michalroszka replied:
:white_check_mark: addressed in 1cb5b960e7626840d729b87dc9e6005494890f12

kvas-damian commented on extensions/wikia/SpecialVideos/SpecialVideosHooks.class.php:
```diff
@@ -0,0 +1,49 @@
+					'src' => wfBlankImgUrl(),
+					'class' => 'sprite addRelatedVideo'
+				],
+				null
```
I believe we can use default value here.

michalroszka replied:
:white_check_mark: addressed in 1cb5b960e7626840d729b87dc9e6005494890f12

kvas-damian commented on extensions/wikia/Wall/WallHooksHelper.class.php:
```diff
@@ -529,86 +529,51 @@ static public function onSkinSubPageSubtitleAfterTitle( $title, &$ptext, &$cssCl
 				&& !empty( $parts[1] )
 				&& mb_strtolower( str_replace( ' ', '_', $parts[1] ) ) === mb_strtolower( $helper->getArchiveSubPageText() )
 			) {
-				// user talk archive
```
Don't we need to handle this case anymore?

onei replied:
I'm not 100% sure what I did here, but this conditional seems very close to the next one with the only difference being how the namespace is checked and that an extra argument is used when creating the edit URL. It also looks like the result of this conditional is immediately overridden by the next one - they set the same keys in the response.

At a guess, I think I spotted the duplication and removed this in favour of the later conditional as it appeared the later one was producing the result anyway.

kvas-damian replied:
I can see 2 differences here:
- we check namespace on different object: https://github.com/Wikia/app/blob/ef9e673f7725829ff0f88d022ca74c8739d33367/extensions/wikia/Wall/WallHooksHelper.class.php#L580-580 - I don't know if it's for purpose or not unfortunately.
- `$userTalkPageTitle` has different value: https://github.com/Wikia/app/blob/ef9e673f7725829ff0f88d022ca74c8739d33367/extensions/wikia/Wall/WallHooksHelper.class.php#L586-586

onei replied:
With regards to the first difference the namespace is always derived from `$wgTitle`, see the value of `$ns` being set [here](https://github.com/Wikia/app/blob/42a351cba9082b2eff9a69ee8b5c92df077d1203/skins/oasis/modules/PageHeaderController.class.php#L181-L201). It's essentially the same condition for both, so either both or neither conditional is executed no matter what.

Are there subpages in NS_WALL other than user talk archives that this should handle or used to handle? That's the only reason I can think of for the two conditions to exist in the first place.

michalroszka replied:
:exclamation: I need to review and re-think it

michalroszka replied:
:white_check_mark: This has been addressed.

kvas-damian commented on extensions/wikia/Wall/WallHooksHelper.class.php:
```diff
@@ -1751,28 +1716,36 @@ static public function onDiffViewHeader( $differenceEngine, $oldRev, $newRev ) {
-			$pageHeaderModule->subtitle = Xml::element( 'a', array( 'href' => $wmRef->getMessagePageUrl() ), wfMessage( 'oasis-page-header-back-to-article' )->text() );
+
+			$pageHeaderModule->title = wfMessage( 'oasis-page-header-diff' )
+				->rawParams( htmlspecialchars( $meta ) )
```
I know it was here before but isn't it equivalent of `params( $meta )`?

michalroszka replied:
:exclamation: i am not so sure about this one, need to confirm that by testing

kvas-damian commented on extensions/wikia/WikiaHubsV3/css/WikiaHubsV3.scss:
```diff
@@ -22,68 +22,46 @@ $wikiahubs-line-height: 16px;
+		// making this display:none causes the bottom border to go above/across
+		// the social links because they're floated instead of using flex
+		// like the default page header
+		visibility: hidden;
```
Maybe we could set height of parent element and use `display: none` here?

michalroszka replied:
! not sure about this one, need to confirm that by testing

onei replied:
Done in [de16f7f](https://github.com/Wikia/app/commit/de16f7f610f2d01ba4f8ace5263b7e4160cbf995)

onei commented on skins/oasis/modules/templates/PageHeader_Index.php:
```diff
@@ -1,145 +1,76 @@
 				</div>
-				<div class="header-column header-tally">
-					<? if ( !empty( $wg->EnablePageShareExt ) ): ?>
-						<div id="PageShareContainer" class="page-share-container">
```
Does the [PageShare JS](https://github.com/Wikia/app/blob/dev/extensions/wikia/PageShare/scripts/PageShare.js#L40) need updating as well due to the case change on the ID?

kvas-damian replied:
Yes, https://github.com/Wikia/app/blob/18f13e4d4956b445932fc25444c9501c0ded10c9/extensions/wikia/PageShare/PageShareController.class.php#L9-9 This module is empty by default and lazy-loaded then.

michalroszka replied:
:white_check_mark: adressed in 9ba716e91dc4ac6bae8ab9f892298649187e4608

kvas-damian commented on includes/Article.php:
```diff
@@ -610,13 +610,24 @@ public function view() {
-							$wgOut->addHTML( $this->viewRedirect( $rt ) );
+							// begin wikia change
+							// VOLDEV-145
+							$wgOut->addHTML( $this->viewRedirect( $rt, $appendSubtitle ) );
```
I'm not sure what we're try to achieve here. Can you give us some background?

onei replied:
There's an issue with links being added to breadcrumbs in inconsistent ways, due to sometimes manipulating what is output in core skins and adding it to a different breadcrumbs  and sometimes outputting it as is - compare the HTML and font-size on [redirect (action=delete)](http://runescape.wikia.com/wiki/KBD?action=delete) to [non-redirect (action=delete)](http://runescape.wikia.com/wiki/King_Black_Dragon?action=delete) with regards to `<- $page` link.

This is actually caused by a weird set of conditions when building the subtitle, see [here](https://github.com/Wikia/app/pull/9609/files?diff=unified#diff-518d114e986a0d2589af415d26592578R322) down to the end of the function. If it was me I'd unify the two subtitles, but I assumed there was a reason it was handled this way originally so I tried to make it more consistent instead.

kvas-damian replied:
Makes sense now.

michalroszka replied:
:white_check_mark:

onei commented on skins/oasis/css/core/PageHeader.scss:
```diff
@@ -44,13 +48,11 @@
 		display: inline;
 		font-size: 26px;
```
One thing I never got around to fixing was the alignment of the edit, talk, etc. buttons and pageshare links with the h1 in the header. It works fine for the max screenwidth, but will be slightly out for the next  breakpoint down as the font-size shrinks from 34px to 28px (26px is never used afaik). Any ideas how to fix it?

kvas-damian replied:
Maybe we could styles in MediaQueries to align with new HTML markup?

michalroszka replied:
:question: does https://github.com/WikiaVolDev/app/commit/4ff5617491d6aa833edc1e3601fbb97897aebcc3#diff-5a1adbc1aca329ef8b292df01c02a86eR314 have anything to do with this one?

onei replied:
Yep, this one's fixed now :)

Or rather, it will be when #9649 is merged into this

michalroszka replied:
:white_check_mark:

kvas-damian commented on skins/oasis/css/core/PageHeader.scss:
```diff
@@ -3,40 +3,44 @@
-	&.separator {
-		border-bottom: 1px solid $color-page-border;
-		padding: 0 0 15px;
+.page-header {
```
When introducing this new selector we changes many styles, eg. this rule is applied on H1 https://github.com/Wikia/app/blob/130ef8aac02b1631af89ea1739bc5cb9aef3bcb3/skins/oasis/css/core/responsive-font-sizing.scss#L67-67
Is it intended?

onei replied:
Sorry, at the time I completely forgot font-size and line-height was set elsewhere. That, and other related styles, need to be updated to use the new class name.

michalroszka replied:
:question: has this been addressed in 829ea998b9369645ba8b2bc5aa96f91c882a8e5e ?

onei replied:
I think it's all sorted between [829ea99](https://github.com/Wikia/app/commit/829ea998b9369645ba8b2bc5aa96f91c882a8e5e) and [4ff5617](https://github.com/Wikia/app/commit/4ff5617491d6aa833edc1e3601fbb97897aebcc3). A few areas I haven't managed to test due to issues with my devbox or not being familiar with how they work to test them:
- Places
- Maps
- VE (Can't get it working for some reason, but it's a very minor fix if it's wrong)
- Ads/Tracking JS
- Scavenger hunt
- Quiz
- Wall (diffs and deleted pages)

I also updated the selectors in the styles for responsive layout, although I'm not sure if they're still in use so didn't do any extra cleanup around that area.

michalroszka replied:
Okay, our dedicated QM will do QA for us.

kvas-damian commented on skins/oasis/css/core/PageHeader.scss:
```diff
@@ -58,228 +60,39 @@
 	.wikia-button,
 	.wikia-menu-button {
-		margin-top: 2px;
+		margin-top: 8px;
```
Because of this rule edit button is not vertically centered to page title.

onei replied:
As I mentioned above, it'll break depending on screen width and does need fixing before merging.

michalroszka replied:
:question: adressed in https://github.com/Wikia/app/pull/9646/files#diff-5a1adbc1aca329ef8b292df01c02a86eR314 perhaps?

onei replied:
Yep, that's fixed now :)

kvas-damian commented on skins/oasis/css/modules/ContributeModule.scss:
```diff
@@ -0,0 +1,46 @@
+@import 'skins/shared/mixins/flexbox';
+
+.contribute-module {
```
We could polish this module when right rail is dropped to the bottom of the page. Currently tally and buttons are displayed one under another...

michalroszka replied:
:question: @kvas-damian is there any immediate action with regard to this one?

onei replied:
Still trying to figure out what's causing this one. It's worth noting however it starts behaving weirdly again if there's >1m pages on a wiki.

onei replied:
Fixed in [63aa6f5](https://github.com/Wikia/app/commit/63aa6f5d3ba86c91b0bf043d6c0c1011ac41892e)

michalroszka replied:
:white_check_mark:

kvas-damian commented on skins/oasis/modules/templates/PageHeader_Index.php:
```diff
@@ -1,145 +1,76 @@
-						<div class="subtitle"><?= $subtitle ?></div>
-					<? endif; ?>
+			// don't show share buttons on Special:Videos as we want the tally instead
+			if ( !empty( $wg->EnablePageShareExt && !$isSpecialVideos ) ) { ?>
```
I believe `!$isSpecialVideos` shouldn't be inside `empty()`

onei replied:
Done :)

kvas-damian commented on skins/oasis/css/modules/ContributeModule.scss:
```diff
@@ -0,0 +1,46 @@
+
+		em {
+			float: left;
+    		font-size: 30px;
```
Please fix indentation

onei replied:
Done :)

kvas-damian commented on skins/oasis/modules/BodyController.class.php:
```diff
@@ -77,286 +111,309 @@ public static function isGridLayoutEnabled() {
+		$railModuleList = [];
+
+		// temp plocation for testing
+		// @todo figure out where this should go
```
Any plans on resolving this TODO?

onei replied:
VOLDEV-145's ticket description said " This module will not be present on main pages or other pages without the standard right rail", but nothing more than that. This was only meant to be temporary until the definition of the standard right rail was clarified, but unfortunately the question was never answered on the ticket, hence the leftover TODO.

kvas-damian replied:
See @macbre comment here: https://github.com/Wikia/app/pull/9609#discussion_r51703694

kvas-damian commented on skins/oasis/modules/BodyController.class.php:
```diff
@@ -77,286 +111,309 @@ public static function isGridLayoutEnabled() {
 		}
 
-		$railModuleList[1440] = array('Ad', 'Index', ['slotName' => 'TOP_RIGHT_BOXAD']);
-		$railModuleList[1435] = array('AdEmptyContainer', 'Index', ['slotName' => 'NATIVE_TABOOLA_RAIL']);
```
Is it safe to remove this ad unit?
Ping @nandy-andy

onei replied:
I think that's just an accidental removal during cleanup.

nandy-andy replied:
Let's bring it back, please. We'll probably remove `NATIVE_TABOOLA_RAIL` one day but `TOP_RIGHT_BOXAD` is quite important for the company :smiley_cat:

onei replied:
Done :)

michalroszka replied:
:white_check_mark: this is fine, I can see both slots in the code.

nandy-andy replied:
Thanks you, Sirs :bow:

macbre commented on skins/oasis/modules/BodyController.class.php:
```diff
@@ -77,286 +111,309 @@ public static function isGridLayoutEnabled() {
+
+		// temp plocation for testing
+		// @todo figure out where this should go
+		$railModuleList[1460] = [ 'Contribute', 'index', null ];
```
The spec in the VOLDEV-145 says

> Small module at the top of the right rail (below ad unit) 

So I guess the index here should be something around 1490+.

onei replied:
Wouldn't 1490+ be above everything in the right rail? I thought it needed to be somewhere between PagesOnWiki (1450) and Ad (1490)?

macbre replied:
Aren't right rail modules rendered based on the index order of `$railModuleList` array? Or is there more magic in Oasis than I think? :)

kvas-damian replied:
I believe it's ordered using those indexes, although on non article pages I see really low indexes. Probably we'll need to make sure that contribute module appears correctly also on other pages.

onei replied:
@macbre I think I've misunderstood what Ad is exactly, hence the confusion. I took it to mean the ad unit at the top of the right rail. If that's not the case then 1490+ would indeed be the ideal value :)

michalroszka replied:
:white_check_mark: this has been addressed

kvas-damian commented on skins/oasis/modules/PageHeaderController.class.php:
```diff
@@ -8,126 +8,121 @@
-		if ( isset( $this->content_actions['rate'] ) ) {
-			$this->action = $this->content_actions['rate'];
-			$this->actionName = 'rate';
+		// @todo does this even work? couldn't find it in production
```
It's defined in https://github.com/Wikia/app/blob/8a3ebc352c13fd80acce2db2b5fe4a38b99b7eaa/extensions/3rdparty/PvX/Rate.php#L30-30

It's not used anymore. Will create cleanup ticket. SUS-131

michalroszka replied:
:white_check_mark:

kvas-damian commented on skins/oasis/modules/PageHeaderController.class.php:
```diff
@@ -161,107 +156,102 @@ private function getCuratedContentButton() {
-
-		/** start of wikia changes @author nAndy */
-		$response = $this->getResponse();
-		if ( $response instanceof WikiaResponse ) {
```
Don't we need this check anymore?

onei replied:
It was only used for hooks which have been altered to only allow what they should, e.g. passing only the necessary variables to them. See `PageHeaderIndexAfterActionButtonPrepared` and `PageHeaderIndexExtraButtons`.

kvas-damian replied:
:+1:

michalroszka replied:
:white_check_mark:

kvas-damian commented on skins/oasis/modules/PageHeaderController.class.php:
```diff
@@ -609,31 +536,32 @@ public function executeCorporate() {
-		$wgMemc->delete( wfMemcKey( 'mOasisRecentRevisions2', $article->getTitle()->getArticleId() ) );
-		return true;
+		// temp styles loader for testing
+		$this->response->addAsset( 'wikiahubs_v3_scss' );
```
Are those styles included here only for debugging?

onei replied:
Yes, I couldn't get the styles to load on my devbox without it. I guessed it was down to missing config, so would someone else be able to check if it works without it?

kvas-damian replied:
Yep, QA have config file set on their application instance, so it should be safe to remove it now.

michalroszka replied:
:white_check_mark: addressed in 9ac5cbfea31e83f796ae6c11a5fe4d62d663a222

kvas-damian commented:
@onei Thanks for all your work here. You had to change a lot in this PR. It's appreciated.

mattdowdell committed:
changes per CR

mattdowdell committed:
change all usages of #WikiaPageHeader and .WikiaPageHeader to their new names

mattdowdell committed:
fix selector on pageshare js for lazy-loading

mattdowdell committed:
clean up CSS a little more, fix forum after conversion to flex, fix print.css for wikiheader (broken by other things but that's for another day)

mattdowdell committed:
remove temp loading of styles

Commit:
Merge branch 'dev' into SUS-28-dev-VOLDEV-145-reapply-changes-by-onei

Commit:
Merge pull request #9646 from WikiaVolDev/SUS-28

SUS-28/VOLDEV-145

mattdowdell committed:
more CR changes
Fix whitespace in PlacesHooks
Set parent element height and display:none; h1 for hubs page header

mattdowdell committed:
remove completed todo

mattdowdell committed:
fix issue with alignment on minimum width breakpoint, also fix modules splitting across multiple columns

Commit:
Merge pull request #9649 from WikiaVolDev/SUS-28

SUS-28 - More CR changes

mattdowdell committed:
more CR fixes
Remove line-height and font-size from pageheader styles so it's obviously not set there
Fix condition for user talk archives/subpages

Commit:
Merge pull request #9653 from WikiaVolDev/SUS-28

SUS-28 - More CR changes (2)

Commit:
Merge branch 'dev' into SUS-28-dev-VOLDEV-145-reapply-changes-by-onei

Commit:
SUS-144 | Disable contribute module on special pages

Commit:
SUS-145 | SUS-147 | Move contribute module under Medrec ad

Commit:
SUS-145 | Fix lazy loaded contribute button

Commit:
SUS-148 | Fix styling Right Rail modules on small screens

Commit:
SUS-28 | Remove unused code

Commit:
SUS-146 | Contribute module - fix layout for big numbers

Commit:
SUS-143 | Fix contribute button tracking

Commit:
SUS-172 | Fix Add a page button under contribute Menu

Commit:
SUS-28 | Center contribute button

Commit:
SUS-171 | Fix edit button on contribute menu

Commit:
Merge branch 'dev' into SUS-28-dev-VOLDEV-145-reapply-changes-by-onei

Commit:
Merge branch 'dev' into SUS-28-dev-VOLDEV-145-reapply-changes-by-onei

 Conflicts:
	extensions/wikia/Forum/ForumHooksHelper.class.php
	extensions/wikia/Wall/WallHooksHelper.class.php

onei commented:
@kvas-damian @mixth-sense As a double check, https://github.com/Wikia/app/commit/e3beec37e5db9c17d29b3a137023f2be75ee02f5 undoes [ `63aa6f5`](https://github.com/Wikia/app/commit/63aa6f5d3ba86c91b0bf043d6c0c1011ac41892e#diff-5a1adbc1aca329ef8b292df01c02a86eR169) which was added due to WikiActivity splitting across multiple columns when no ads were present (I think it happened in Chrome, can't remember about others). I notice the structure of the module has changed slightly, so is that issue no longer valid?

ghost commented:
@onei - we'll get this one through QA again.

Commit:
Merge branch 'dev' into SUS-28-dev-VOLDEV-145-reapply-changes-by-onei

Commit:
Merge branch 'dev' into SUS-28-dev-VOLDEV-145-reapply-changes-by-onei

Commit:
Merge branch 'dev' into SUS-28-dev-VOLDEV-145-reapply-changes-by-onei

Commit:
Merge branch 'dev' into SUS-28-dev-VOLDEV-145-reapply-changes-by-onei
